### PR TITLE
[ISSUE #3102 common] add ArrayUtils、ArrayUtilsTest

### PR DESCRIPTION
**Pull request by zzq1314zll**
[ISSUE #3102] Improve the tool class in Nacos-common, replace common-lang3

- [x] Add ArrayUtils/ArrayUtilsTest
- [ ] Add method for StringUtils/StringUtilsTest
- [ ] Add BooleanUtils/BooleanUtilsTest
- [ ] Add NumberUtils/NumberUtilsTest
- [ ] Add RandomUtils/RandomUtilsTest
- [ ] Add TypeUtils/TypeUtilsTest
- [ ] Add ObjectUtils/ObjectUtilsTest
- [ ] Add ClassUtils/ClassUtilsTest
- [ ] Add SystemUtils/SystemUtilsTest
This is the API using Common-lang3 in Nacos.Add them step by step

### CODE REVIEW

File: common/src/main/java/com/alibaba/nacos/common/utils/ArrayUtils.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.common.utils;
+
+import java.lang.reflect.Array;
+
+/**
+ * Array utils. Copy from common-lang3.
```
Comment (KomachiSion) [inline common/src/main/java/com/alibaba/nacos/common/utils/ArrayUtils.java:22]: If copy from apache2 license, need keep the original copyright header. And keep the original author.

Then add some common to describe what's your changes.

**Review by KomachiSion — CHANGES_REQUESTED** [review]

**Comment by realJackSun** [comment]
Thx for contribution, But only two methods of ArrayUtils is used in Nacos, maybe you can remove the redundant ones.

**Comment by zzq1314zll** [comment]
> Thx for contribution, but you should take care two points:
> 1 As @KomachiSion says, keep the original copyright header, the original author, and describe your changes.
> 2 Only two methods of ArrayUtils is used in Nacos, please remove the redundant ones.
> image

Thank you for your suggestion, I will do as you say

**Comment by zzq1314zll** [comment]
> Thx for contribution, but you should take care two points:
> 1 As @KomachiSion says, keep the original copyright header, the original author, and describe your changes.
> 2 Only two methods of ArrayUtils is used in Nacos, please remove the redundant ones.
> image

Thank you for your suggestion, I will do as you say

**Pull request closed without merging**